src/test/java/com/cegeka/academy/service/ChallengeServiceTest.java

```diff
@@ -64,36 +79,52 @@ public void init(){
         challengeDTO.setStartDate(new Date());
         challengeDTO.setPoints(22);
         challengeDTO.setStatus("status");
+
+        challenge = new Challenge();
+
+        challenge.setCreator(user);
+        challenge.setStartDate(new Date(System.currentTimeMillis()));
+        challenge.setEndDate(new Date(System.currentTimeMillis()));
+        challenge.setStatus("Activa");
+        challenge.setDescription("Nimic");
+        challenge.setPoints(10);
+        challenge.setChallengeCategory(challengeCategory);
+        defaultChallengeId = challengeRepository.save(challenge).getId();
     }
 
     @Test
     public void testSaveChallenge(){
 
-      challengeService.saveChallenge(challengeDTO);
+       challengeService.saveChallenge(challengeDTO);
+
+       int lastItemIndex = (int)challengeRepository.count() - 1;
 
-       assertThat(challengeRepository.findAll().get(0).getCreator()).isEqualTo(new UserMapper().userDTOToUser(challengeDTO.getCreator()));
```

Comment:
De ce ai modificat aici? Testele folosesc ca baza de date H2, iar cand se trece de la un test la altul nu se pastreaza ce a fost anterior salvat. Eu in test salvez un singur challenge si este normal sa fie primul element din lista returnata de findAll().

Reply:
Am mai salvat eu un challenge in setUp() de care aveam nevoie ca sa testez stergerea. Avand in vedere ca in testul tau se face save(), challenge-ul salvat o sa fie ultimul din lista si il preiau asa. E posibil ca pe viitor sa fie nevoie sa mai adaugam poate inca un Challenge in setUp, in felul asta nu trebuie modificat indexul de fiecare data.

Reply:
ok, ai dreptate

Reply:
Va rog sa modificați implementarea astfel încât să nu fiți nevoiți să faceți astfel de artificii cu calcularea unor indecși. Prin urmare, trebuie să ștergeti tot ce adăugați înainte sau in timpul unui test după ce ați făcut asserturile